meson.build

```diff
@@ -0,0 +1,277 @@
+        error('CoreText was enabled explicitly, but required headers or frameworks are missing.')
+    endif
+endif
+
+if get_option('require-system-font-provider') and not fontconfig and not directwrite and not coretext
+    error('''Either DirectWrite (on Windows), CoreText (on OSX), or Fontconfig
+(Linux, other) is required. If you really want to compile without
+a system font provider, set -Drequire-system-font-provider=false''')
+endif
+
+# ASM
+
+asm_args = []
+
+# Architecture check
+if host_machine.cpu_family() == 'x86'
+    bittype = '32'
+    asm_args += '-DARCH_X86_64=0'
+    cpu_family = 'x86'
+elif host_machine.cpu_family() == 'x86_64'
+    bittype = '64'
+    asm_args += '-DARCH_X86_64=1'
+    cpu_family = 'x86'
+elif host_machine.cpu_family() == 'aarch64' and host_system == 'darwin'
+    bittype = '64'
+    cpu_family = 'arm64'
+endif
+
+if cpu_family == 'x86'
+    nasm = find_program('nasm', required: get_option('asm'))
+    enable_asm = nasm.found()
+    use_nasm = true
+elif cpu_family == 'arm64'
+    enable_asm = true
+    use_nasm = false
+else
+    enable_asm = false
+    use_nasm = false
+endif
+
+# NASM version check
+if enable_asm and use_nasm
+    # This can be cleaned up once 0.64 is available, see https://github.com/xclaesse/libass/commit/cfd3d9971bd585939b97409c0aac79cab3dd2da5
+    nasm_r = run_command(nasm, '-v', check: true)
+    out = nasm_r.stdout().strip().split()
+    if out[1].to_lower() == 'version'
+        if out[2].version_compare('< 2.10')
+            warning('nasm is too old (found @0@); ASM functions are disabled.'.format(out[2]))
+            enable_asm = false
+        endif
+    else
+        warning('Unexpected nasm version string: @0@; disabling.'.format(nasm_r.stdout()))
+        enable_asm = false
+    endif
+endif
+
+if get_option('asm').enabled() and (not enable_asm)
+    error('ASM functions were enabled but are unavailable.')
+elif (not enable_asm) and use_nasm and (not get_option('asm').disabled())
+    warning('Install nasm-2.10 or later for a significantly faster libass build.')
+endif
+
+# System check
+if enable_asm
+    if host_system == 'windows'
+        asm_args += ['-f', 'win' + bittype]
+        if bittype == '32'
+            asm_args += '-DPREFIX'
+        endif
+    elif host_system == 'darwin'
+        if cpu_family == 'arm64'
+            asm_args += '-DPREFIX'
+        else
+            asm_args += ['-f', 'macho' + bittype, '-DPREFIX', '-DSTACK_ALIGNMENT=16']
+        endif
+    elif host_system == 'linux' or host_system == 'sunos' or host_system == 'haiku' or host_system == 'gnu'
+        asm_args += ['-f', 'elf' + bittype, '-DSTACK_ALIGNMENT=16']
+    elif host_system == 'dragonfly' or host_system.endswith('bsd')
+        asm_args += ['-f', 'elf' + bittype]
+    else
+        error('Please contact libass upstream to figure out if ASM support for your platform can be added. In the meantime, disabling.')
+        enable_asm = false
+    endif
+endif
```

Comment:
If I’m reading the `asm_arg` handling correctly, the flags set here will also be added on non-x86 arches, eg.g aarch64 Linux. However, those flags are NASM-specific and likely break *something* on other arches. *(NASM exclusively deals with the x86 family, on all other arches we have to get along with the less featurefull GNU/LLVM assembler)*

Imho with the aarch64 addition the assembly handling here is now more convoluted than in `configure.ac`. You don't need to exactly follow `configure.ac` if you have better ideas, but could you perhaps see if this can be reorganised to be clearer?

Reply:
Yeah, I missed that when adding the GNU asm handling. Should be fixed now, also with some hopefully clearer logic for the flags.